src/main/java/org/owasp/validator/html/scan/ASHTMLSerializer.java

```diff
@@ -67,4 +70,14 @@ public void endElementIO(String namespaceURI, String localName,
 			_printer.flush();
 	}
 
+	@Override
+	protected String escapeURI(String uri) {
+    	String originalURI = uri;
+		try {
+			printEscaped(uri);
+		} catch (IOException e) {
+			logger.error("URI escaping failed for value: " + originalURI);
+		}
+		return "";
```

Comment:
Is this method supposed to ALWAYS return an empty string? Seems strange to me.

Reply:
It is. On the HTMLSerializer, that function is supposed to returned the URL with escaped quotes and nothing else. I'm reusing `printEscaped` which does the encoding we want and internally "prints" the result in the XML that is being built. If I return a non-empty string, that string is also "printed" because the call to `escapeURI` is something like `print(escapeURI())`. I mentioned this on our emails, it may be explained better there.

Reply:
Can you add all the necessary explanation to this code as comments so it stands alone? Otherwise future reviewers will have the same concerns.

Reply:
I've done a separate PR with the comment. I tried to push directly to 1.6.4 and had git issues, but I'm kinda in a hurry and wanted to have this solved fast.